dealership/contracts.py

```diff
@@ -1,12 +1,44 @@
 class Contract(object):
-    pass
+    def __init__(self, vehicle, customer):
+        self.vehicle = vehicle
+        self.customer = customer
 
 
 class BuyContract(Contract):
     def __init__(self, vehicle, customer, monthly_payments):
-        pass
+        super(BuyContract, self).__init__(vehicle, customer)
+        self.monthly_payments = monthly_payments
+
+    def total_value(self):
+        sale_price = self.vehicle.sale_price()
+        interest = self.vehicle.interest_multiplier
+
+        total = sale_price + (interest * sale_price *
+                              self.monthly_payments / 100)
+
+        if self.customer.is_employee():
+            return total * 0.9
+        return total
+
+    def monthly_value(self):
+        return self.total_value() / self.monthly_payments
 
 
 class LeaseContract(Contract):
     def __init__(self, vehicle, customer, length_in_months):
-        pass
+        super(LeaseContract, self).__init__(vehicle, customer)
+        self.length_in_months = length_in_months
+
+    def total_value(self):
+        sale_price = self.vehicle.sale_price()
+        lease_multiplier = self.vehicle.lease_multiplier
+
+        total = sale_price + \
+            (sale_price * lease_multiplier / self.length_in_months)
+
+        if self.customer.is_employee():
+            return total * 0.9
+        return total
+
+    def monthly_value(self):
+        return self.total_value() / self.length_in_months
```

Comment:
Your code is neatly formatted, and you do a good job of making sure the files meet PEP-8 guidelines.